### ISSUE TITLE
nrf: neopixel chains of certain lengths cause hard fault

### ISSUE THREAD

**Issue by dhalbert** [adafruit/circuitpython#1962]
Found by @bmeisels in #circuitpython discord. Reduced to simple test case:
```
>>> import neopixel,board
>>> px = neopixel.NeoPixel(board.D6, 1)
>>> px[0] = (5,5,5)
>>> px.deinit()
>>> px = neopixel.NeoPixel(board.D6, 2)
>>> px[0] = (5,5,5)
[hard crash]
```
In general, it fails for `neopixel.NeoPixel(board.D6, <n>)` for `n` = 2 through 5, then it works for 6 and above.

**Comment by jerryneedell** [comment]
@dhalbert I ran examples discussed in discord on a feather_m4 express with no problems so this is an nrf52840 issue

**Comment by jerryneedell** [comment]
never-mind -- I see the nrf tag

**Comment by tannewt** [comment]
Fixed by #1963
